ee/desktop/runner/runner.go

```diff
@@ -313,32 +309,128 @@ func (r *DesktopUsersProcessesRunner) Ping() {
 	level.Debug(r.logger).Log("msg", "runner processSpawningEnabled:%s", strconv.FormatBool(enabled))
 }
 
-// writeMenuFile writes menu data to a shared file for user processes to access
-func (r *DesktopUsersProcessesRunner) writeMenuFile(data menu.MenuData) error {
+// writeSharedFile writes data to a shared file for user processes to access
+func (r *DesktopUsersProcessesRunner) writeSharedFile(path string, data any) error {
 	menuBytes, err := json.Marshal(data)
 	if err != nil {
-		return fmt.Errorf("failed to marshal menu json: %w", err)
+		return fmt.Errorf("failed to marshal json: %w", err)
 	}
 
-	menuPath := r.menuPath()
-	statusFile, err := os.Create(menuPath)
+	file, err := os.Create(path)
 	if err != nil {
-		return fmt.Errorf("creating desktop menu file: %w", err)
+		return fmt.Errorf("creating file: %w", err)
 	}
 
-	if err := os.Chmod(menuPath, 0644); err != nil {
+	if err := os.Chmod(path, 0644); err != nil {
 		return fmt.Errorf("os.Chmod: %w", err)
 	}
 
-	defer statusFile.Close()
-	_, err = io.Copy(statusFile, bytes.NewReader(menuBytes))
+	defer file.Close()
+	_, err = io.Copy(file, bytes.NewReader(menuBytes))
```

Comment:
why `io.Copy`, and not `file.Write`

Reply:
👍 Changes to `file.Write` so I can avoid having to create a reader.